packages/race-cli/src/reporters/aggregate-console-reporter.ts

```diff
@@ -79,20 +112,79 @@ export class AggregateConsoleReporter extends BaseRacepointReporter {
     this._settings = options;
   }
 
-  process = async (results: LighthouseResultsWrapper) => {
-    Object.values(LightHouseAuditKeys).forEach((value) => {
-      this._collectedData[value].push(results.lhr.audits[value].numericValue);
+  process = async (results: UserFlowResultsWrapper) => {
+    results.steps.forEach((step: UserFlowStep, i: number) => {
+      const stepKey = `step_${i}`;
+
+      // Initialize a new empty step if it doesn't exist
+      if (!this._stepDataCollection[stepKey]) {
+        this._stepDataCollection[stepKey] = emptyStep(
+          step.name || `Step #${i}`
+        );
+      }
+      const activeStep = this._stepDataCollection[stepKey];
+
+      Object.values(LightHouseAuditKeys).forEach((value) => {
+        if (step.lhr.audits[value]) {
+          activeStep[value].push(step.lhr.audits[value].numericValue);
+        }
+      });
     });
   };
 
   async finalize(cacheStats?: CacheMetricData): Promise<void> {
```

Comment:
this function is bordering on "too big", imo.

I like the Math Operation objects, but you could, for example, extract these two out into a 'default MathBook' (also, great name)